prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -2036,6 +2050,18 @@ subroutine initProgramVariables(input, env)
         call move_alloc(dftd4, dispersion)
       end if
       cutOff%mCutOff = max(cutOff%mCutOff, dispersion%getRCutOff())
+    #:if WITH_MBD
+      elseif (allocated(input%ctrl%dispInp%mbdInp)) then
+        tManyBodyDisp = .true.
+        allocate (mbDispersion)
+        associate (inp => input%ctrl%dispInp%mbdInp)
+            inp%calculate_forces = tForces
+            inp%atom_types = speciesName(species0)
+            inp%coords = coord0
+            if (tPeriodic) inp%lattice_vectors = latVec
```

Comment:
Why can't this part be populated in the parser?

Reply:
Can it? I thought some of these quantities might not be known yet at the parsing time. Happy to change it.